db/migrate/20190518154841_add_permissions_to_users.rb

```diff
@@ -0,0 +1,19 @@
+class AddPermissionsToUsers < ActiveRecord::Migration[5.2]
+  def up
+    add_column :users, :permissions, :string, array: true, default: []
+    add_index :users, :permissions, using: 'gin'
+
+    execute "UPDATE users SET permissions = ARRAY['view', 'annotate', 'import'];"
+    execute "UPDATE users SET permissions = array_append(permissions, 'manage_users') WHERE admin = TRUE;"
+
+    remove_column :users, :admin
```

Comment:
Having been bitten before by problematic schema updates, I’d like to separate the addition of the `permissions` column from the removal of the `admin` column, and ship the former *before* the rest of this PR so that we don’t have any problems with the code and DB being out of sync during rolling updates or if we need to revert a change.

Reply:
👍 I removed the removal of the `admin` column from the migration